src/main/java/io/github/moderngamingworlds_mods/woodenutilities/common/recipes/WoodcutterRecipeBuilder.java

```diff
@@ -21,22 +22,38 @@
     public List<String> requiredMods = new ArrayList<>();
     public List<String> ingredients = new ArrayList<>();
     public String result;
-    public int count;
+    public int count = 1;
 
     public WoodcutterRecipeBuilder(ResourceLocation loc) {
         this.loc = loc;
     }
 
-    public static void from(WoodcutterRecipe recipe, Consumer<FinishedRecipe> recipeConsumer) {
-        String path = "woodcutter/".concat(Objects.requireNonNull(recipe.result.getItem().getRegistryName()).getPath());
-        var woodcutterRecipeBuilder = new WoodcutterRecipeBuilder(new ResourceLocation(WoodenUtilities.getInstance().modid, path));
-        woodcutterRecipeBuilder.requiredMods.addAll(recipe.requiredMods);
-        for (ItemStack item : recipe.ingredient.getItems()) {
-            woodcutterRecipeBuilder.ingredients.add(Objects.requireNonNull(item.getItem().getRegistryName()).toString());
-        }
-        woodcutterRecipeBuilder.result = recipe.result.getItem().getRegistryName().toString();
-        woodcutterRecipeBuilder.count = recipe.result.getCount();
-        recipeConsumer.accept(woodcutterRecipeBuilder);
+    public static WoodcutterRecipeBuilder create(ResourceLocation loc) {
+        return new WoodcutterRecipeBuilder(loc);
+    }
+
+    public WoodcutterRecipeBuilder requiredMod(String modId) {
+        this.requiredMods.add(modId);
+        return this;
+    }
+
+    public WoodcutterRecipeBuilder ingredient(Item item) {
```

Comment:
I'd say we should use `ItemLike`. This way, you could also use blocks/`BlockItem`s